modules/util/portal-tools-rest-builder-test-test/src/testIntegration/java/com/liferay/portal/tools/rest/builder/test/resource/v1_0/test/TestEntityResourceTest.java

```diff
@@ -188,12 +204,153 @@ public void testPatchTestEntity() throws Exception {
 		assertValid(getChildTestEntity2);
 	}
 
+	@Test
+	public void testPostImportTask() throws Exception {
+		ChildTestEntity1 childTestEntity1 = new ChildTestEntity1();
+
+		childTestEntity1.setName(
+			StringUtil.toLowerCase(RandomTestUtil.randomString()));
+		childTestEntity1.setProperty1(
+			StringUtil.toLowerCase(RandomTestUtil.randomString()));
+		childTestEntity1.setType(TestEntity.Type.create("ChildTestEntity1"));
+
+		ChildTestEntity2 childTestEntity2 = new ChildTestEntity2();
+
+		childTestEntity2.setName(
+			StringUtil.toLowerCase(RandomTestUtil.randomString()));
+		childTestEntity2.setProperty2(
+			StringUtil.toLowerCase(RandomTestUtil.randomString()));
+		childTestEntity2.setType(TestEntity.Type.create("ChildTestEntity2"));
+
+		ChildTestEntity3 childTestEntity3 = new ChildTestEntity3();
+
+		childTestEntity3.setName(
+			StringUtil.toLowerCase(RandomTestUtil.randomString()));
+		childTestEntity3.setType(TestEntity.Type.create("ChildTestEntity3"));
+
+		Page<TestEntity> page = testEntityResource.getTestEntitiesPage(null);
+
+		long totalCount = page.getTotalCount();
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.batch.engine.internal." +
+					"BatchEngineImportTaskExecutorImpl",
+				LoggerTestUtil.ERROR)) {
+
+			String invalidTypeId = StringUtil.toLowerCase(
+				RandomTestUtil.randomString());
+
+			JSONObject jsonObject = waitForFinish(
+				"COMPLETED",
+				HTTPTestUtil.invokeToJSONObject(
+					JSONUtil.putAll(
+						JSONFactoryUtil.createJSONObject(
+							ChildTestEntity1SerDes.toJSON(childTestEntity1)),
+						JSONUtil.put(
+							"name",
+							StringUtil.toLowerCase(
+								RandomTestUtil.randomString())
+						).put(
+							"type", invalidTypeId
+						),
+						JSONFactoryUtil.createJSONObject(
+							ChildTestEntity2SerDes.toJSON(childTestEntity2)),
+						JSONUtil.put(
+							"name",
+							StringUtil.toLowerCase(
+								RandomTestUtil.randomString())),
+						JSONFactoryUtil.createJSONObject(
+							ChildTestEntity3SerDes.toJSON(childTestEntity3))
+					).toString(),
+					StringBundler.concat(
+						"headless-batch-engine/v1.0/import-task",
+						"/com.liferay.portal.tools.rest.builder.test.dto.v1_0.",
+						"TestEntity?importStrategy=ON_ERROR_CONTINUE"),
+					Http.Method.POST));
+
+			page = testEntityResource.getTestEntitiesPage(null);
+
+			Assert.assertEquals(totalCount + 3, page.getTotalCount());
+
+			assertContains(childTestEntity1, (List<TestEntity>)page.getItems());
+			assertContains(childTestEntity2, (List<TestEntity>)page.getItems());
+			assertContains(childTestEntity3, (List<TestEntity>)page.getItems());
+
+			jsonObject = HTTPTestUtil.invokeToJSONObject(
+				null,
+				"headless-batch-engine/v1.0/import-task/" +
+					jsonObject.getLong("id"),
+				Http.Method.GET);
+
+			JSONArray failedItemsJSONArray = jsonObject.getJSONArray(
+				"failedItems");
+
+			Assert.assertEquals(2, failedItemsJSONArray.length());
+			Assert.assertTrue(
```

Comment:
I think we can use here the containsThat expression, the message in case of failure is probably more clear:

```java
Assert.assertThat(
			failedItemsJSONArray.getJSONObject(
					0
				).getString(
					"message"
				),
			CoreMatchers.containsString(Could not resolve type id '" + invalidTypeId +
						"' as a subtype"));
```

Reply:
Also we could use:

```java
JSONUtil.getValueAsString(failedItemsJSONArray, "JSONObject/0", "Object/message")
```

(not sure if this is more clear or not, but at least we avoid chaining xD)

Reply:
ok, so let's leave as it is, it is ok!